### PR TITLE
Arrivals Door Replacement + New Door Type

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
This adds a new var (protectedDoor) for airlocks, allowing modders to protect doors against power outages, screwdrivers, AI operation, welders, and emags. 
Airlocks leading into the arrivals shuttle are now replaced with an arrivals-specific door that uses this. The doors are still destructible and deconstructible with an RCD.
Imgur Image

## Why It's Good For The Game
Killing inside of the arrivals shuttle is universally against rules and frowned upon.
In the event players enter the less fortunate stations without power, they no longer need to break glass in order to enter the station, which inadvertently depressurizes the arrivals shuttle and kills newcomers. 
Low-level griefing like hacking the arrivals shuttle doors is now deterred but not completely impossible.

## Changelog
:cl:
add: Airlocks can now be marked as protected against power outages and sabotage.
tweak: Arrivals airlocks now use this function and open with or without power.
add: New arrivals airlock sprite.
/:cl:

### CODE REVIEW

**Comment by Cenrus** [comment]
why is this up to vote lole

**Comment by ike709** [comment]
👎 I'm not opposed to the change, but these should have a unique icon or some other way of visually signifying that they're protected.

**Comment by ike709** [comment]
Hell maybe just make the door light blue and have it on all of the time or something.

**Comment by jlsnow301** [comment]
> 👎 I'm not opposed to the change, but these should have a unique icon.

I thought about making a white airlock with "WELCOME" on it, and can probably PR the icon once I get around to it

**Comment by TheNeoGamer42** [comment]
this is needed, stop having it up to a vote and accept it now damnet

**Comment by jlsnow301** [comment]
New sprites added.

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -928,7 +937,7 @@
 						security_level = AIRLOCK_SECURITY_PLASTEEL_O
 					return
 	if(C.tool_behaviour == TOOL_SCREWDRIVER)
-		if(panel_open && detonated)
+		if(panel_open && detonated || protectedDoor)
```
Comment (ike709) [inline code/game/machinery/doors/airlock.dm]:
```suggestion
		if((panel_open && detonated) || protectedDoor)
```

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -983,7 +992,7 @@
 /obj/machinery/door/airlock/try_to_weld(obj/item/weldingtool/W, mob/user)
 	if(!operating && density)
 		if(user.a_intent != INTENT_HELP)
-			if(!W.tool_start_check(user, amount=0))
+			if(!W.tool_start_check(user, amount=0) || protectedDoor)
```
Comment (ike709) [inline code/game/machinery/doors/airlock.dm]:
```suggestion
			if(protectedDoor || !W.tool_start_check(user, amount=0))
```

Comment (ike709) [reply code/game/machinery/doors/airlock.dm]: Also it needs player feedback.

**Comment by MCterra10** [comment]
Just port what tg did, and add an emergency release catch to the doors, so that they can be opened without tools when unpowered.

**Comment by ike709** [comment]
@MCterra10 I'm not sure if I'm a fan of that change tbh.

**Comment by qwertyquerty** [comment]
Protected airlocks shouldn't use any of the station power when operated. Make sure to add in a check for that.

File: code/game/machinery/doors/airlock.dm [hunk 3]
```diff
@@ -94,6 +94,7 @@
 
 	var/air_tight = FALSE	//TRUE means density will be set as soon as the door begins to close
 	var/prying_so_hard = FALSE
+	var/protectedDoor = FALSE // Protects the door against any form of power outage, AI control, screwdrivers and welders.
```
Comment (qwertyquerty) [inline code/game/machinery/doors/airlock.dm]: Change all instances to protected_door

Comment (jlsnow301) [reply code/game/machinery/doors/airlock.dm]: done

**Comment by jlsnow301** [comment]
> Protected airlocks shouldn't use any of the station power when operated. Make sure to add in a check for that.

Done


> Just port what tg did, and add an emergency release catch to the doors, so that they can be opened without tools when unpowered.

I was not aware of this change actually, is this ALL doors?

**Pull request merged**